model/project_parser.go

```diff
@@ -48,33 +48,33 @@ const LoadProjectError = "load project error(s)"
 // configuration YAML. It implements the Unmarshaler interface
 // to allow for flexible handling.
 type ParserProject struct {
-	Id                 string                     `yaml:"_id" bson:"_id"` // should be the same as the version's ID
-	ConfigUpdateNumber int                        `yaml:"config_number,omitempty" bson:"config_number,omitempty"`
-	Enabled            bool                       `yaml:"enabled,omitempty" bson:"enabled,omitempty"`
-	Stepback           bool                       `yaml:"stepback,omitempty" bson:"stepback,omitempty"`
-	PreErrorFailsTask  bool                       `yaml:"pre_error_fails_task,omitempty" bson:"pre_error_fails_task,omitempty"`
-	OomTracker         bool                       `yaml:"oom_tracker,omitempty" bson:"oom_tracker,omitempty"`
-	BatchTime          int                        `yaml:"batchtime,omitempty" bson:"batchtime,omitempty"`
-	Owner              string                     `yaml:"owner,omitempty" bson:"owner,omitempty"`
-	Repo               string                     `yaml:"repo,omitempty" bson:"repo,omitempty"`
-	RemotePath         string                     `yaml:"remote_path,omitempty" bson:"remote_path,omitempty"`
-	Branch             string                     `yaml:"branch,omitempty" bson:"branch,omitempty"`
-	Identifier         string                     `yaml:"identifier,omitempty" bson:"identifier,omitempty"`
-	DisplayName        string                     `yaml:"display_name,omitempty" bson:"display_name,omitempty"`
-	CommandType        string                     `yaml:"command_type,omitempty" bson:"command_type,omitempty"`
+	Id                 *string                    `yaml:"_id" bson:"_id"` // should be the same as the version's ID
```

Comment:
Are these changes backwards compatible?

Reply:
One way to look into this would be to put your changes in staging, create a patch, and then put HEAD in staging (i.e. remove your changes) to verify that your patch doesn't break.